agent/consul/usagemetrics/usagemetrics.go

```diff
@@ -137,4 +150,66 @@ func (u *UsageMetricsReporter) runOnce() {
 	}
 
 	u.emitServiceUsage(serviceUsage)
+
+	servers, clients := u.memberUsage()
+	u.emitMemberUsage(servers, clients)
+}
+
+func (u *UsageMetricsReporter) memberUsage() (int, map[string]int) {
+	if u.getMembersFunc == nil {
+		return 0, nil
+	}
+
+	mems := u.getMembersFunc()
+	if len(mems) <= 0 {
+		u.logger.Warn("cluster reported zero members")
+		return 0, nil
+	}
+
+	servers := 0
+	clients := make(map[string]int)
+
+	for _, m := range mems {
+		if m.Status != serf.StatusAlive {
+			continue
+		}
+
+		switch m.Tags["role"] {
+		case "node":
+			clients[m.Tags["segment"]]++
```

Comment:
Nice! I am glad it worked out for you where you didn't need a nested for loop over all segments and then over members of the segment.